packages/core-components/src/components/calendar/calendar.tsx

```diff
@@ -0,0 +1,192 @@
+    this.selectedDay = undefined;
+  };
+
+  private setSelectedDate() {
+    if (this.selectedDay !== undefined)
+      this.selectedDate =
+        this.selectedDay +
+        '.' +
+        (this.selectedMonth + 1) +
+        '.' +
+        this.selectedYear;
+    this.b2bSelected.emit({
+      selectedDate: new Date(this.selectedDate),
+    });
+    setTimeout(() => {
+      this.setFocusOnCloseIcon();
+    }, 100);
+  }
+
+  private setFocusOnCloseIcon() {
+    let closeIcon = this.host.shadowRoot.querySelector(
+      '.b2b-close-icon',
+    ) as HTMLDivElement;
+    if (closeIcon !== null) {
+      closeIcon.focus();
+    }
+  }
+
+  getPreviousMonth = () => {
+    if (this.selectedMonth === 0) {
+      this.setCurrentMonth(11);
+      this.setCurrentYear(this.selectedYear - 1);
+    } else {
+      this.setCurrentMonth(this.selectedMonth - 1);
+    }
+  };
+  getNextMonth = () => {
+    if (this.selectedMonth === 11) {
+      this.setCurrentMonth(0);
+      this.setCurrentYear(this.selectedYear + 1);
+    } else {
+      this.setCurrentMonth(this.selectedMonth + 1);
+    }
+  };
+
+  render() {
+    return (
+      <Host>
+        <div class="b2b-calendar">
+          <div class="b2b-calendar-label">{this.label}</div>
+          <div
+            class={{
+              'b2b-calender-input-wrapper': true,
+              'b2b-calender-input-wrapper--opened': this.showCalendar,
+            }}
+            tabindex={0}
+            onClick={this.showHideCalendar}
+            onKeyDown={event => {
+              if (event.key === 'Enter') {
+                this.showHideCalendar();
+              }
+            }}>
+            <div class="b2b-calendar-selected-date">{this.selectedDate}</div>
+            <div class="b2b-icons">
+              {this.selectedDate && (
+                <div
+                  tabIndex={0}
+                  onClick={() => {
+                    this.clearDateInput();
+                    this.showHideCalendar();
+                  }}
+                  class="b2b-close-icon"
+                  onKeyDown={event => {
+                    if (event.key === 'Enter') {
+                      this.clearDateInput();
+                      this.showHideCalendar();
+                    }
+                  }}>
+                  <b2b-icon
+                    icon="b2b_icon-close"
+                    class="b2b-icon"
+                    aria-label="clear input"
+                    clickable={true}></b2b-icon>
+                </div>
+              )}
+
+              <div class="b2b-calender-icon" tabindex={0}>
+                <b2b-icon
+                  class="b2b-icon"
+                  aria-label={
+                    this.showCalendar ? 'close calender' : 'open calender'
+                  }
+                  icon="b2b_icon-event"
+                  clickable={true}></b2b-icon>
+              </div>
+            </div>
+          </div>
+        </div>
+        {this.showCalendar && (
```

Comment:
Personally, I would modify visibility via CSS here (display: block / display: none) as opposed to conditional rendering.

The current solution rerenders all dates again whenever the calendar is opened, which is less performant.